airbyte_cdk/connector_builder/test_reader/helpers.py

```diff
@@ -0,0 +1,590 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import json
+from copy import deepcopy
+from json import JSONDecodeError
+from typing import Any, Dict, List, Mapping, Optional
+
+from airbyte_cdk.connector_builder.models import (
+    AuxiliaryRequest,
+    HttpRequest,
+    HttpResponse,
+    StreamReadPages,
+    StreamReadSlices,
+)
+from airbyte_cdk.models import (
+    AirbyteLogMessage,
+    AirbyteMessage,
+    OrchestratorType,
+    TraceType,
+)
+from airbyte_cdk.models import Type as MessageType
+from airbyte_cdk.sources.utils.slice_logger import SliceLogger
+from airbyte_cdk.sources.utils.types import JsonType
+from airbyte_cdk.utils.datetime_format_inferrer import DatetimeFormatInferrer
+from airbyte_cdk.utils.schema_inferrer import (
+    SchemaInferrer,
+)
+
+from .types import LOG_MESSAGES_OUTPUT_TYPE
+
+
+class Parsers:
+    @staticmethod
+    def airbyte_message_to_json(message: AirbyteMessage) -> Optional[Dict[str, JsonType]]:
+        """
+        Converts an AirbyteMessage to a JSON dictionary if its type is LOG.
+
+        This function attempts to parse the 'log' field of the given AirbyteMessage when its type is MessageType.LOG.
+        If the parsed JSON object exists but is not a dictionary, a ValueError is raised. If the message is not of type LOG,
+        the function returns None.
+
+        Parameters:
+            message (AirbyteMessage): The AirbyteMessage instance containing the log data.
+
+        Returns:
+            Optional[Dict[str, JsonType]]: The parsed log message as a dictionary if the message type is LOG, otherwise None.
+
+        Raises:
+            ValueError: If the parsed log message is not a dictionary.
+        """
+        if Conditions.is_log_message(message):
```

Comment:
I'm not following where `Conditions` is coming from. I don't see it imported or declared in the file. Can you clarify?

Reply:
Nevermind. Declared below. I see.

Reply:
The module design here feels a bit odd to me. I know it is common in Java to have helper classes for shared functionality, but I think in this case, the helper functions could stand on their own since they are all static methods. 

Optionally, you could still divide up into `...helpers/parsers.py` and `...helpers/conditions.py` if you want to keep the function scope small.

Wdyt?

Reply:
It was before today's commit)) funny to admit. Ok, I'll revert it back.

Reply:
Done.